bigframes/core/reshape/__init__.py

```diff
@@ -125,9 +125,22 @@ def cut(
         if isinstance(bins, pd.IntervalIndex):
             as_index: pd.IntervalIndex = bins
             bins = tuple((bin.left.item(), bin.right.item()) for bin in bins)
-        else:
+        elif len(list(bins)) == 0:
+            raise ValueError("`bins` iterable should have at least one item")
+        elif isinstance(list(bins)[0], tuple):
             as_index = pd.IntervalIndex.from_tuples(list(bins))
             bins = tuple(bins)
+        elif pd.api.types.is_number(list(bins)[0]):
+            if len(list(bins)) < 2:
+                raise ValueError(
```

Comment:
pandas doesn't seem to have this restriction. can we not have this exception?

Reply:
I looked into this quite a bit this morning, but it looks like we need more support for the struct type in order to be able to create the column of NaN structs that this case requires.